README.md

```diff
@@ -51,192 +67,323 @@ and we recommend the one that begins with "https"). Let's call this the base URL
+
+// Run handlerFunction for all button presses inside an unfurl attachment
+slackMessages.action({ unfurl: true, type: 'button' }, handlerFunction);
+
+// This function is discussed in "Responding to actions" below
+function handlerFunction() {
+}
 ```
 
-These options should allow you to match actions in any way that makes sense for your app. Keep in
-mind that only the first handler that matches an action will be invoked.
+#### Responding to actions
+
+Slack requires your app to respond to actions in a timely manner so that the user isn't blocked.
+The adapter helps your app respond correctly and on time.
+
+For most actions (button presses and menu selections), a response is simply an updated message to
+replace the one where the interaction occurred. Your app can return a message (or a Promise for a
+message) from the handler. **We recommend that apps at least remove the interactive elements from
+the message in the response** so that users don't get confused (for example, click the same button
+twice). Find details about the format for a message in the docs for
+[message](https://api.slack.com/docs/interactive-message-field-guide#message).
 
-### Asynchronous responses
+The handler will receive a `payload` which describes the interaction a user had with a message.
+Find more details about the structure of `payload` in the docs for
+[buttons](https://api.slack.com/docs/message-buttons#responding_to_message_actions) and
+[menus](https://api.slack.com/docs/message-menus#request_url_response).
 
-Sometimes you can't prepare a response to an action synchronously, especially in node where all I/O
-is evented and you should not block. In these situations we recommend that you at least consider
-updating the message to remove the interactive elements (often times actions aren't meant to trigger
-more than once for an app). The adapter will invoke your handler with a `respond()` function as the
-second argument so you can update the message asynchronously.
+If your app defers some work asynchronously (like querying another API or using a database), you can
+continue to update the message using the `respond()` function that is provided to your handler.
+
+**Example button handler:**
 
 ```javascript
-slackMessages.action('my_callback', (payload, respond) => {
-  // Do some asynchronous work (returns a Promise)
-  doExpensiveThing()
-    // Call respond() with a JSON object that represents a replacement message
-    .then(formatMessage)
-    .then(respond)
-    // Set `replace_original` to `false` if the message is not meant to replace the original.
-    .catch((error) => respond({ text: error.message, replace_original: false }));
-
-  // Remove the interactivity from the message and update the content to acknowledge the interaction
-  const updatedMessage = acknowledgeInteraction(payload.original_message);
-  return updatedMessage;
+slackMessages.action('welcome_agree_button', (payload, respond) => {
```

Comment:
this is great, a much clearer example of how the respond method works, and I think it's useful to have the respond method be invoked in the HelloWorld action - in nearly any case you'll want to have an 'all good' and an 'error' version of your response.